vumi/transports/vumi_bridge/vumi_bridge.py

```diff
@@ -152,101 +142,22 @@ def get_auth_headers(self):
-            config.redis_manager)
-        self.retries = 0
-        self.delay = config.initial_delay
-        self.reconnect_call = None
-        self.client = StreamingClient(self.agent_factory)
-        self.connect_api_clients()
-
-    def teardown_transport(self):
-        if self.reconnect_call:
-            self.reconnect_call.cancel()
-            self.reconnect_call = None
-        self.continue_trying = False
-        self.disconnect_api_clients()
-
-    def connect_api_clients(self):
-        self.message_client = self.client.stream(
-            TransportUserMessage, self.handle_inbound_message,
-            log.error, self.get_url('messages.json'),
-            headers=Headers(self.get_auth_headers()),
-            on_connect=self.reset_reconnect_delay,
-            on_disconnect=self.reconnect_api_clients)
-        self.event_client = self.client.stream(
-            TransportEvent, self.handle_inbound_event,
-            log.error, self.get_url('events.json'),
-            headers=Headers(self.get_auth_headers()),
-            on_connect=self.reset_reconnect_delay,
-            on_disconnect=self.reconnect_api_clients)
-
-    def reconnect_api_clients(self, reason):
-        self.disconnect_api_clients()
-        if not self.continue_trying:
-            log.msg('Not retrying because of explicit request')
-            return
-
-        config = self.get_static_config()
-        self.retries += 1
-        if (config.max_retries is not None
-                and (self.retries > config.max_retries)):
-            log.warning('Abandoning reconnecting after %s attempts.' % (
-                self.retries))
-            return
-
-        self.delay = min(self.delay * config.factor,
-                         config.max_reconnect_delay)
-        if config.jitter:
-            self.delay = random.normalvariate(self.delay,
-                                              self.delay * config.jitter)
-        log.msg('Will retry in %s seconds' % (self.delay,))
-        self.reconnect_call = self.clock.callLater(self.delay,
-                                                   self.connect_api_clients)
-
-    def reset_reconnect_delay(self):
-        config = self.get_static_config()
-        self.delay = config.initial_delay
-        self.retries = 0
-        self.reconnect_call = None
-        self.continue_trying = True
-
-    def disconnect_api_clients(self):
-        self.message_client.disconnect()
-        self.event_client.disconnect()
-
+    def update_status(self, **kw):
+        '''Publishes a status if it is not a repeat of the previously
+        published status.'''
+        if self.status_detect.check_status(**kw):
+            yield self.publish_status(**kw)
 
-class GoConversationTransport(GoConversationClientTransport):
+    def add_status_bad_req(self, component):
+        return self.update_status(
+            status='down', component=component, type='bad_request',
+            message='Bad request received')
```

Comment:
We need to set `type` and `message` more appropriately for the different error / success cases (e.g. `Bad request received` is not correct when we're failing to make requests).

Maybe this means that `add_status_good_req` and `add_status_bad_req` should be removed? Or maybe we should have a pair of these methods for each kind of status update?